auraed/src/runtime/cells/cell.rs

```diff
@@ -91,17 +140,28 @@ impl Cell {
         Ok(())
     }
 
-    pub fn kill_executable(
+    pub fn stop_executable(
         &mut self,
         exe_name: &ExecutableName,
-    ) -> Result<ExitStatus> {
-        if let Some(exe) = self.executables.remove(exe_name) {
-            Ok(exe.kill()?)
+    ) -> Result<ExitStatus, CellError> {
+        if let Some(mut exe) = self.executables.remove(exe_name) {
+            match exe.kill() {
+                Ok(exit_status) => Ok(exit_status),
+                Err(e) => {
+                    // Failed to kill, put it back in cache
+                    let _ = self.executables.insert(exe_name.clone(), exe);
```

Comment:
instead of remove/fail/insert.. how about get/kill/remove?

ie get it, try to kill it, remove it if successful